core/arch/arm/include/mm/core_memprot.h

```diff
@@ -45,14 +45,13 @@
 
 /* memory atttributes */
 enum buf_is_attr {
-	CORE_MEM_SEC,
+	CORE_MEM_CACHED,
+	CORE_MEM_EXTRAM,
+	CORE_MEM_NSEC_SHM,
 	CORE_MEM_NON_SEC,
+	CORE_MEM_SEC,
 	CORE_MEM_TEE_RAM,
 	CORE_MEM_TA_RAM,
-	CORE_MEM_NSEC_SHM,
-	CORE_MEM_EXTRAM,
-	CORE_MEM_INTRAM,
-	CORE_MEM_CACHED,
```

Comment:
GCC has a flag `-Wenum-compare` which is supposed to catch errors like these. It's supposed to be enabled via the `-Wall` flag so I'm a bit surprised we're having this error. Perhaps it doesn't cover assignments?

I agree with Etienne, this may decrease some optimizations. I don't have a good idea for a bulletproof fix, the only I can think of is getting rid of one of the enums but that's a bit complicated and out of scope here.